Java/README.md

```diff
@@ -0,0 +1,687 @@
+
+1.  All static imports in a single block.
+2.  All non-static imports in a single block.
+
+If there are both static and non-static imports, a single blank line separates the two blocks. There are no other blank lines between import statements.
+
+Within each block the imported names appear in ASCII sort order. (**Note:** this is not the same as the import _statements_ being in ASCII sort order, since '.' sorts before ';'.)
+
+#### 3.3.4 No static import for classes
+
+Static import is not used for static nested classes. They are imported with normal imports.
+
+### 3.4 Class declaration
+
+#### 3.4.1 Exactly one top-level class declaration
+
+Each top-level class resides in a source file of its own.
+
+#### 3.4.2 Ordering of class contents
+
+The order you choose for the members and initializers of your class can have a great effect on learnability. However, there's no single correct recipe for how to do it; different classes may order their contents in different ways.
+
+What is important is that each class uses **_some_ logical order**, which its maintainer could explain if asked. For example, new methods are not just habitually added to the end of the class, as that would yield "chronological by date added" ordering, which is not a logical ordering.
+
+##### 3.4.2.1 Overloads: never split
+
+When a class has multiple constructors, or multiple methods with the same name, these appear sequentially, with no other code in between (not even private members).
+
+## 4 Formatting
+
+**Terminology Note:** _block-like construct_ refers to the body of a class, method or constructor. Note that, by Section 4.8.3.1 on [array initializers](#s4.8.3.1-array-initializers), any array initializer _may_ optionally be treated as if it were a block-like construct.
+
+### 4.1 Braces
+
+#### 4.1.1 Braces are used where optional
+
+Braces are used with `if`, `else`, `for`, `do` and `while` statements, even when the body is empty or contains only a single statement.
+
+#### 4.1.2 Nonempty blocks: K & R style
+
+Braces follow the Kernighan and Ritchie style ("[Egyptian brackets](http://www.codinghorror.com/blog/2012/07/new-programming-jargon.html)") for _nonempty_ blocks and block-like constructs:
+
+*   No line break before the opening brace.
+*   Line break after the opening brace.
+*   Line break before the closing brace.
+*   Line break after the closing brace, _only if_ that brace terminates a statement or terminates the body of a method, constructor, or _named_ class. For example, there is _no_ line break after the brace if it is followed by `else` or a comma.
+
+Examples:
+
+
+return () -> {
+  while (condition()) {
+    method();
+  }
+};
+
+return new MyClass() {
+  @Override public void method() {
+    if (condition()) {
+      try {
+        something();
+      } catch (ProblemException e) {
+        recover();
+      }
+    } else if (otherCondition()) {
+      somethingElse();
+    } else {
+      lastThing();
+    }
+  }
+};

```

Comment:
Minor: consider indenting these or using ``` so that it is properly formatted as code. (This applies to all code samples in this file.)

Reply:
Fixed ! wasnt sure if this syntax works for java, but seems yes .. thanks!